libcxx/include/__config

```diff
@@ -647,6 +647,14 @@ typedef __char32_t char32_t;
 #define _LIBCUDACXX_IS_CONSTANT_EVALUATED(...) __builtin_is_constant_evaluated(__VA_ARGS__)
 #endif
 
+// TODO: Builtin is not constexpr, this can be tested with an integral_constant
+//       but _only_ in C++11 mode. (fix for NVCC+clang-9 and up)
+#if defined(__CUDACC__)
+#if __cplusplus < 201402L
+#undef _LIBCUDACXX_IS_CONSTANT_EVALUATED
+#endif
+#endif
```

Comment:
Can we do this without an `#undef` by moving it into the NVCC configuration section, and only defining `_LIBCUDACXX_IS_CONSTANT_EVALUATED` if not Clang 9 and not C++11?

Reply:
Also I'm confused, does this affect GCC too? If so we should have the comment say that. The test indicates it affects GCC 9 and 10 and Clang 9 and 10?